network-policies/readme.adoc

```diff
@@ -0,0 +1,175 @@
+
+```
+/ # wget -S http://http-echo.ns-1.svc.cluster.local/test --post-data '{"message":"hello"}' -O test
+Connecting to http-echo.ns-1.svc.cluster.local (100.64.161.56:80)
+wget: can't connect to remote host (100.64.61.223): Connection timed out
+```
+
+=== Create network policy with a rule
+
+We will now delete the NetworkPolicy that we just created, and create a new NetworkPolicy with a rule defined.  If you `cat templates/allow-network-policy.yaml` you will see the following rule defined:
+
+  spec:
+    podSelector:
+      matchLabels:
+        app: http-echo
+    ingress:
+      - from:
+        - podSelector:
+            matchLabels:
+              app: busybox
+
+The rule above is stating that for every pod that has the label `app: http-echo` defined, allow access to it from pods that have the label `app: busybox` defined.
+
+Run the following in `shell 3` to remove the deny all by default rule, and replace with the above allow rule:
+
+  $ kubectl delete netpol deny-all-by-default --namespace=ns-1
+  networkpolicy "deny-all-by-default" deleted
+  $ kubectl create -f templates/allow-network-policy.yaml --namespace=ns-1
+  networkpolicy "allow" created
+
+If we repeat the following command in `shell 2`, the call should still timeout and fail (again, you can press CTRL-C to quit after 10 seconds rather than waiting for the full timeout to occur):
+
+```
+/ # wget -S http://http-echo.ns-1.svc.cluster.local/test --post-data '{"message":"hello"}' -O test
+Connecting to http-echo.ns-1.svc.cluster.local (100.64.161.56:80)
+wget: can't connect to remote host (100.64.61.223): Connection timed out
+```
+
+Why is this still failing even after creating a rule?  It is failing because we configured the rule so that only pods with the label `app: busybox` are authorized to call pods with the label `app: http-echo`.  Let's go ahead and label our `busybox` pod on `shell 3`:
+
+  / # kubectl label po --selector=run=busybox --namespace=ns-1 app=busybox
+  pod "busybox" labeled
+
+Repeating the test in `shell 2` again should now be successful:
+
+```
+/ # wget -S http://http-echo.ns-1.svc.cluster.local/test --post-data '{"message":"hello"}' -O test
+Connecting to http-echo.ns-1.svc.cluster.local (100.64.161.56:80)
+  HTTP/1.1 200 OK
+  X-Powered-By: Express
+  Content-Type: application/json; charset=utf-8
+  Content-Length: 536
+  ETag: W/"218-xgvU8WZSN+2SEyOX6Q2R/AhLuRM"
+  Date: Thu, 26 Oct 2017 02:15:32 GMT
+  Connection: close
+
+test                100% |*************************************************************************************************|   534   0:00:00 ETA
+```
+
+== Teardown
+
+Remove all the resources and the namespace using the command:
+
+  \ # kubectl delete ns ns-1
+  namespace "ns-1" deleted
```

Comment:
Have you verified that this section would work with both Calico and Weavenet?

Reply:
I can run through it with Calico, but ideally we'd get someone more familiar with Calico to vet the changes.